File: RemoveElement_27.java
class Solution {
    public static int removeElement(int[] nums, int val) {
        int l = 0;
        for(int r = 0; r < nums.length; r++) {
            if(nums[r] != val) {
                nums[l++] = nums[r];
            }
        }
        return l;
    }
    
    public static void main(String args[]) {
        int[] nums = {3,3,2};
        System.out.println(removeElement(nums,3));
    }
}

File: RemoveElement_27.py

class Solution(object):
    def removeElement(self, nums, val):
        k = 0
        for vl in nums:
            if vl != val:
                nums[k] = vl
                k = k + 1
        return k
if __name__ == '__main__':
    solu = Solution()
    nums = [3,2,2,3]
    print(solu.removeElement(nums, 3))
